hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/JpaMetamodelImpl.java

```diff
@@ -493,6 +500,34 @@ public void processJpa(
 			for ( EmbeddableDomainType<?> embeddable : context.getEmbeddableTypeSet() ) {
 				this.jpaEmbeddableDescriptorMap.put( embeddable.getJavaType(), embeddable );
 			}
+			Stream.concat(
+					context.getEntityTypesByEntityName().values().stream(),
+					Stream.concat(
+							context.getMappedSuperclassTypeMap().values().stream(),
+							context.getEmbeddableTypeSet().stream()
+					)
+			).forEach( managedDomainType -> {
+				managedDomainType.visitAttributes( persistentAttribute -> {
+					if ( persistentAttribute.getJavaType().isEnum() ) {
```

Comment:
Seems like this will break in cases of de-typed domain models (aka "map mode").  Depends how we end up deciding to handle that in building the JPA metamodel.  Regardless, its easy enough to handle here simply by checking for `persistentAttribute.getJavaType() == null`

Reply:
Nice catch, thanks. I adapted it.